### PR TITLE
Package geos 3.4.2

### PR DESCRIPTION
**Pull request by pretyman**
This package is interesting because:
* creates a script file in the `bin` dir, which won't be `lipo`'ed
* needs some extra linking flags on Android

### CODE REVIEW

**Comment by pretyman** [comment]
I pushed a simple example to trigger the build.

File: cmake/find/Findgeos.cmake [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  if(NOT GEOS_C_LIBRARY)
+    message(FATAL_ERROR "GEOS_C_LIBRARY not found!")
+  endif()
+  add_library("geos::c" UNKNOWN IMPORTED)
+  set_target_properties(
+      "geos::c"
+      PROPERTIES
+        IMPORTED_LOCATION "${GEOS_C_LIBRARY}"
+  )
+  add_dependencies(
```
Comment (ruslo) [inline cmake/find/Findgeos.cmake]: add_dependencies make sense only for building targets. Usually dependency set by CMake automatically (like in target_link_libraries) and needed only in very special cases when it's quite hard to find out that automatically (like dependency between several ExternalProject_Add targets). If you want to link `geos` target to every target that use `geos::s` you can use IMPORTED_LINK_INTERFACE_LIBRARIES: https://github.com/ruslo/hunter/commit/c783edc6cf36b709bdd6b5526aeb4a149dfe4e60

**Comment by ruslo** [comment]
> creates a script file in the bin dir, which won't be lipo'ed

https://github.com/ruslo/hunter/issues/218

**Comment by ruslo** [comment]
> needs some extra linking flags on Android

https://github.com/ruslo/hunter/issues/219

File: cmake/find/Findgeos.cmake (hunk 1 above)
Comment (pretyman) [inline cmake/find/Findgeos.cmake]: ooops, thanks, I got mixed up with that one.

**Pull request merged**